src/components/Search/index.tsx

```diff
@@ -105,20 +111,41 @@ const handleSearchResult = async (
 const Search: FC<{
   content?: string
   hasButton?: boolean
+  truncateTypeHash?: boolean
   onEditEnd?: () => void
-}> = memo(({ content, hasButton, onEditEnd }) => {
+}> = memo(({ content, hasButton, onEditEnd, truncateTypeHash }) => {
   const isMobile = useIsMobile()
   const { t } = useTranslation()
+  const searchByType = cacheService.get(SEARCH_TYPE_KEY) || 'id'
+  const [isSearchByNames, setIsSearchByNames] = useState(searchByType === 'name')
+  const [searchByNameResults, setSearchByNameResults] = useState<UdtQueryResult[] | null>(null)
   const history = useHistory()
-  const SearchPlaceholder = t('navbar.search_placeholder')
   const [searchValue, setSearchValue] = useState(content || '')
-  const [placeholder, setPlaceholder] = useState(SearchPlaceholder)
+  const placeholder = isSearchByNames ? t('navbar.search_by_name_placeholder') : t('navbar.search_placeholder')
   const inputElement = useRef<HTMLInputElement>(null)
 
-  // update input placeholder when language change
-  useEffect(() => {
-    setPlaceholder(SearchPlaceholder)
-  }, [SearchPlaceholder])
+  const toggleSearchType = () => {
+    const newIsSearchByNames = !isSearchByNames
+    const searchTypePersistValue = newIsSearchByNames ? 'name' : 'id'
+    cacheService.set(SEARCH_TYPE_KEY, searchTypePersistValue)
+    setIsSearchByNames(newIsSearchByNames)
+    setSearchByNameResults(null)
+  }
+
+  const debouncedSearchByName = useMemo(
+    () =>
+      debounce(
+        (queryString: string) => {
```

Comment:
Here it is suggested to implement it above with react-query, and only debounce the refetch function provided by react-query, so that some features of react-query can be utilized (such as discarding the previous unfinished request).